light-client-js/src/lib.rs

```diff
@@ -25,46 +25,50 @@ use wasm_bindgen::{prelude::*, JsValue};
 /// Check whether a given untrusted block can be trusted.
 #[wasm_bindgen]
 pub fn verify(untrusted: &JsValue, trusted: &JsValue, options: &JsValue, now: &JsValue) -> JsValue {
-    let result = deserialize_params(untrusted, trusted, options, now).map(
-        |(untrusted, trusted, options, now)| {
-            let verifier = ProdVerifier::default();
-            verifier.verify(
-                untrusted.as_untrusted_state(),
-                trusted.as_trusted_state(),
-                &options,
-                now,
-            )
-        },
-    );
-    JsValue::from_serde(&result).unwrap()
+    let result = deserialize_params(
+        untrusted.clone(),
+        trusted.clone(),
+        options.clone(),
+        now.clone(),
+    )
```

Comment:
I wonder if the signature of `verify` can be changed to take the arguments by value.